src/meshpy/four_c/function_utility.py

```diff
@@ -77,3 +77,26 @@ def create_linear_interpolation_function(
 
     variable_string = create_linear_interpolation_string(t, values, variable_name="var")
     return Function(f"{function_type} var\n" + variable_string)
+
+
+def ensure_length_of_function_array(function_array, length: int = 3):
+    """Performs size check of a function array and appends the function array to the given length, if a list with only one item is provided.
+    Args:
+        function_array: list with functions
+        length: expected length of function array
+
+    Returns:
+        function_array: list with functions with provided length
+    """
+    if function_array is None:
+        raise ValueError("Provided array is empty.")
+
+    # extend items of function automatically if it is only provided once
+    if len(function_array) == 1:
+        return function_array * length
+
+    if len(function_array) != length:
+        raise ValueError(
+            f"The function array must have length {length} not {len(function_array)}."
+        )
```

Comment:
I think `function_array is None` gives the wrong result.

```suggestion
    if len(function_array) == 0:
        raise ValueError("Provided array is empty.")

    # extend items of function automatically if it is only provided once
    if len(function_array) == 1:
        return function_array * length

    if len(function_array) != length:
        raise ValueError(
            f"The function array must have length {length} not {len(function_array)}."
        )
```

Reply:
this check had to be introduced when i tried to add the types `List[Union[Int,Function]]`, I think it makes sense to remove this completly.